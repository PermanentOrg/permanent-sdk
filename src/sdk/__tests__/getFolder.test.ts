import nock from 'nock';
import { getFolder } from '..';

describe('getFolder', () => {
  it('should return a Folder', async () => {
    const folderPage = nock('https://api.permanent.local')
      .get('/api/v2/folder')
      .query({
        'folderIds[]': '7457',
      })
      .replyWithFile(
        200,
        `${__dirname}/fixtures/getFolder/stelaFolder.json`,
        {
          'Content-Type': 'application/json',
        },
      );

    const firstChildrenPage = nock('https://api.permanent.local')
      .get('/api/v2/folder/7457/children')
      .query({
        pageSize: 20,
      })
      .replyWithFile(
        200,
        `${__dirname}/fixtures/getFolder/stelaChildrenPageOne.json`,
        {
          'Content-Type': 'application/json',
        },
      );

    const secondChildrenPage = nock('https://api.permanent.local')
      .get('/api/v2/folder/7457/children')
      .query({
        pageSize: 20,
        cursor: '1575064',
      })
      .replyWithFile(
        200,
        `${__dirname}/fixtures/getFolder/stelaChildrenPageTwo.json`,
        {
          'Content-Type': 'application/json',
        },
      );

    const folder = await getFolder(
      {
        bearerToken: '12345',
        stelaBaseUrl: 'https://api.permanent.local/api/v2',
      },
      {
        folderId: 7457,
      },
    );

    expect(folderPage.isDone()).toBe(true);
    expect(firstChildrenPage.isDone()).toBe(true);
    expect(secondChildrenPage.isDone()).toBe(true);
    expect(folder).toMatchSnapshot();
  });

  it('should throw an error if stela does not return the folder', async () => {
    const folderPage = nock('https://api.permanent.local')
      .get('/api/v2/folder')
      .query({
        'folderIds[]': '7457',
      })
      .replyWithFile(
        200,
        `${__dirname}/fixtures/getFolder/stelaNoFolder.json`,
        {
          'Content-Type': 'application/json',
        },
      );

    const call = getFolder(
      {
        bearerToken: '12345',
        stelaBaseUrl: 'https://api.permanent.local/api/v2',
      },
      {
        folderId: 7457,
      },
    );
    await expect(call).rejects.toThrow();
    expect(folderPage.isDone()).toBe(true);
  });
});
