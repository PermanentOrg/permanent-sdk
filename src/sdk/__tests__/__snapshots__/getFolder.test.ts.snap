// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getFolder should return a Folder 1`] = `
{
  "archiveRecords": [
    {
      "createdAt": 2022-07-19T15:26:31.000Z,
      "displayDate": 2022-06-01T18:44:43.000Z,
      "displayName": "face-in-clouds",
      "fileSystemCompatibleName": "face-in-clouds.gif",
      "files": [],
      "id": 512,
      "status": "status.generic.ok",
      "type": "type.record.image",
      "updatedAt": 2022-07-19T15:30:23.000Z,
    },
  ],
  "createdAt": 2022-07-19T14:00:45.000Z,
  "displayDate": 2022-06-01T18:44:43.000Z,
  "fileSystemCompatibleName": "testing",
  "folders": [
    {
      "archiveRecords": [],
      "createdAt": 2022-07-19T15:26:40.000Z,
      "displayDate": 2022-07-19T15:26:40.000Z,
      "fileSystemCompatibleName": "Internal folder",
      "folders": [],
      "id": 7458,
      "name": "Internal folder",
      "size": 0,
      "updatedAt": 2022-07-19T15:27:06.000Z,
    },
  ],
  "id": 7457,
  "name": "testing",
  "size": 0,
  "updatedAt": 2022-07-19T15:30:22.000Z,
}
`;

exports[`getFolder should return a Folder when DisplayDT is null 1`] = `
{
  "archiveRecords": [
    {
      "createdAt": 2022-07-19T15:26:31.000Z,
      "displayDate": 2022-06-01T18:44:43.000Z,
      "displayName": "face-in-clouds",
      "fileSystemCompatibleName": "face-in-clouds.gif",
      "files": [],
      "id": 512,
      "status": "status.generic.ok",
      "type": "type.record.image",
      "updatedAt": 2022-07-19T15:30:23.000Z,
    },
  ],
  "createdAt": 2022-07-19T14:00:45.000Z,
  "displayDate": undefined,
  "fileSystemCompatibleName": "testing",
  "folders": [
    {
      "archiveRecords": [],
      "createdAt": 2022-07-19T15:26:40.000Z,
      "displayDate": 2022-07-19T15:26:40.000Z,
      "fileSystemCompatibleName": "Internal folder",
      "folders": [],
      "id": 7458,
      "name": "Internal folder",
      "size": 0,
      "updatedAt": 2022-07-19T15:27:06.000Z,
    },
  ],
  "id": 7457,
  "name": "testing",
  "size": 0,
  "updatedAt": 2022-07-19T15:30:22.000Z,
}
`;
